saleor/tests/e2e/checkout/discounts/promotions/test_promotion_applied_on_checkout_with_specific_subtotal.py

```diff
@@ -0,0 +1,144 @@
+
+
+@pytest.mark.e2e
+def test_promotion_applied_on_checkout_with_specific_subtotal_CORE_2133(
+    e2e_staff_api_client,
+    e2e_not_logged_api_client,
+    shop_permissions,
+    permission_manage_product_types_and_attributes,
+    permission_manage_discounts,
+    permission_manage_orders,
+):
+    permissions = [
+        *shop_permissions,
+        permission_manage_product_types_and_attributes,
+        permission_manage_discounts,
+        permission_manage_orders,
+    ]
+    assign_permissions(e2e_staff_api_client, permissions)
+
+    (
+        channel_id,
+        channel_slug,
+        shipping_method_id,
+        warehouse_id,
+        promotion_id,
+        discount_value,
+        order_predicate_subtotal_value,
+    ) = prepare_promotion(e2e_staff_api_client)
+    (
+        _product_id,
+        product_variant_id,
+        product_variant_price,
+    ) = prepare_product(e2e_staff_api_client, warehouse_id, channel_id, variant_price=7)
+
+    # Step 1 - Create checkout with 5 product variants
+    lines = [{"variantId": product_variant_id, "quantity": 5}]
+    data = checkout_create(
+        e2e_not_logged_api_client,
+        lines,
+        channel_slug,
+        email="testEmail@example.com",
+        set_default_billing_address=True,
+        set_default_shipping_address=True,
+    )
+    checkout_id = data["id"]
+    total = data["totalPrice"]["gross"]["amount"]
+    subtotal = data["subtotalPrice"]["gross"]["amount"]
+    assert data["billingAddress"] is not None
+    assert data["shippingAddress"] is not None
+    assert checkout_id is not None
+    assert data["lines"][0]["variant"]["id"] == product_variant_id
+    assert data["discount"]["amount"] == 0.00
+    assert subtotal != order_predicate_subtotal_value
+
+    # Step 2 - Add shipping method
+    checkout_data = checkout_delivery_method_update(
+        e2e_not_logged_api_client,
+        checkout_id,
+        shipping_method_id,
+    )
+    shipping_price = checkout_data["shippingPrice"]["gross"]["amount"]
+    assert checkout_data["deliveryMethod"]["id"] == shipping_method_id
+
+    # Step 3 - Update checkout lines to only 3 product variants
+    lines = [{"variantId": product_variant_id, "quantity": 3}]
+    checkout_lines = checkout_lines_update(
+        e2e_not_logged_api_client, checkout_id, lines
+    )
+    discounted_subtotal = checkout_lines["checkout"]["subtotalPrice"]["gross"]["amount"]
+    total_gross = checkout_lines["checkout"]["totalPrice"]["gross"]["amount"]
+    undiscounted_subtotal = checkout_lines["checkout"]["lines"][0][
+        "undiscountedTotalPrice"
+    ]["amount"]
+    discounted_subtotal = round(float(undiscounted_subtotal - discount_value), 2)
+    discounted_total = discounted_subtotal + shipping_price
+    assert checkout_lines["checkout"]["discount"]["amount"] == discount_value
+    assert total_gross != total
+    assert total_gross == discounted_total
+    assert undiscounted_subtotal == float(order_predicate_subtotal_value)
```

Comment:
We can also check line prices - undiscounted and discounted unit and total prices, to see if the discount is visible there as well.